### PR TITLE
WIP: PullRequestResource

### PR DESCRIPTION
**Pull request by dlorenc**
This is not ready for review! It is meant to accompany #778 as a draft/POC and [this design doc](https://docs.google.com/document/d/1MXy-AOZmrgNusX710MgeERin_RkjNhnl0P5e5-ROhuc/edit).

### CODE REVIEW

**Comment by bobcatfish** [comment]
v exciting @dlorenc ! could you include in this somewhere (e.g. PR description and/or comments in `pullrequest-init/main.go`) a brief description of what the interface is here? e.g. what files will be provided via the download, and what functionality is available for upload?

**Comment by abayer** [comment]
cc @rawlingsj @ccojocar @pmuir - would be interested to hear your thoughts from a JX perspective.

File: cmd/pullrequest-init/main.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2018 The Knative Authors
```
Comment (hrishin) [inline cmd/pullrequest-init/main.go]: do we still want to continue with `Knative` copyrights?

**Comment by vdemeester** [comment]
Closing this one in favor of #895

**Pull request closed without merging**